### ISSUE TITLE
Rendering particles as points

### ISSUE THREAD

**Issue by darkwater** [djeedai/bevy_hanabi#161]
Instead of rendering quads, would it be possible to have particles rendered as points? As in, `PrimitiveTopology::PointList`

This would always render each particle as exactly one pixel in size. I bet this would have performance gains, but also particles would no longer scale based on distance to camera, which matches the style I'm going for in my game. Combined with HDR and bloom you can still get good looking particles.

(As a bonus, maybe even render as lines, using velocity? Just a thought)

**Comment by djeedai** [comment]
Hi,

Yes we can add a modifier to render particles with a constant camera-space size, that shouldn't be too difficult. However just to note that my (limited) understanding of GPUs tells me you won't get any performance difference because GPUs only draw triangles anyway (`PointList` is a convenience lie, the driver (or GPU directly?) generates fixed-size quads for you I think).

For the "line" case using velocity, that's called trails or stripes and that's a tiny bit involved (requires multiple particles per trail following each other with a custom motion integration logic) but it's very much on my radar of things to add yes.

**Comment by bushrat011899** [comment]
Please correct me if I'm wrong, as I've only been using this library for a week so far. But to my knowledge, there isn't a way to rotate the quads currently used as particles, which means on 3D my effects can "disappear" when a camera looks edge-on at the effect. Allowing the use of points I believe would resolve this problem I'm having.

**Comment by djeedai** [comment]
Yes the `BillboardModifier` does that. And again unless a GPU expert corrects me there's no such thing as "point rendering" on modern GPUs it's just small quads, so there's no point trying to special case this.

**Comment by darkwater** [comment]
The reason I see to implement this, is that the small quads are 1x1 pixel in screenspace, instead of being scaled based on how far away they are is from the camera.

**Comment by djeedai** [comment]
Yes, we do need to add a `ScreenSizeModifier` for sure, so I want to keep this issue open for tracking that. But that is not really "point rendering" (although you can achieve that with a fixed size of 1px).

**Comment by djeedai** [comment]
This should be fixed using the new `screen_space_size` property of either `SetSizeModifier` or `SizeOverLifetimeModifier`.
